lib/assessment.js

```diff
@@ -29,7 +46,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  var ary2 = [];
+  for (var i = 0; i < ary2.length; i++){
+    ary2[] = defaultsFunction(i);
```

Comment:
In this example, `defaultsFunction` is a function, so that part is right. The syntax is still incorrect for adding to an array, though.